cub/block/block_discontinuity.cuh

```diff
@@ -182,15 +182,15 @@ private:
             T                       (&preds)[ITEMS_PER_THREAD],         ///< [out] Calling thread's predecessor items
             FlagOp                  flag_op)                            ///< [in] Binary boolean flag predicate
         {
-            preds[ITERATION] = input[ITERATION - 1];
-
-            flags[ITERATION] = ApplyOp<FlagOp>::FlagT(
-                flag_op,
-                preds[ITERATION],
-                input[ITERATION],
-                (linear_tid * ITEMS_PER_THREAD) + ITERATION);
-
-            Iterate<ITERATION + 1, MAX_ITERATIONS>::FlagHeads(linear_tid, flags, input, preds, flag_op);
+            #pragma unroll
+            for (int i=ITERATION; i < MAX_ITERATIONS; ++i) {
```

Comment:
Actually, it is both `0` (only for FlagTails) and `1` (only for FlagHeads).

Reply:
Should init value be dependent on call to `FlagTails or FlagHeads` or be a template argument? 

Thanks!

Reply:
As you said, both `ITERATION` and `MAX_ITERATIONS` are fixed for these methods:
```bash
:cat block_discontinuity.cuh | rg "Iterate<1" | sort | uniq
        Iterate<1, ITEMS_PER_THREAD>::FlagHeads(linear_tid, head_flags, input, preds, flag_op);
:cat block_discontinuity.cuh | rg "Iterate<0" | sort | uniq
        Iterate<0, ITEMS_PER_THREAD - 1>::FlagTails(linear_tid, tail_flags, input, flag_op);
```
These parameters depend on the `ITEMS_PER_THREAD` that are known in the methods. In my opinion, this creates extra brain load for readers. If you have time and @allisonvacanti doesn't object, let's remove `ITERATION` and `MAX_ITERATIONS`. Otherwise, I'll do it in a separate PR later.

Reply:
I'll be happy to update. wdyt @allisonvacanti?

Reply:
Sounds good to me. Either approach is an improvement, and if we can simplify code by eliminating template params, that's even better :)

Reply:
Will eliminate the params as suggested by @senior-zero. Thanks!